bip-cc.md

```diff
@@ -0,0 +1,207 @@
+validation, prevent Merkle tree weaknesses and avoid duplicate transactions without
+[bip-0030][BIP30] validation.
+
+# Motivation
+
+This proposal addresses a number of long standing vulnerabilities and weaknesses in the Bitcoin
+protocol. Bundling those fixes together allows to overcome the fixed cost of deploying a Bitcoin
+soft fork.
+
+The timewarp bug permits a majority hashrate attacker to arbitrarily increase the block rate,
+allowing them to steal block subsidy from future miners, and increase validation costs to nodes that
+have to deal with the increased average transaction rate. By strategically setting the block
+timestamp, the timewarp bug lets miners bring down the difficulty to its minimum within 38 days of
+starting the attack. The existence of this bug not only significantly empowers a 51% attacker, but
+also makes it notably harder to reason about miners' incentives. It could indeed be in the interest
+of short-sighted miners as well as short-sighted users to exploit this vulnerability in a small
+enough proportion to increase the block rate without fatally hurting the network, as the effectively
+increased block space would - all others things equal - bring fee rates down for users.
+
+Specially crafted blocks may be expensive to process, with validation times ranging from a few
+minutes up to more than a hour on lower end devices. Long block validation times are a nuisance to
+users, increasing the cost to independently fully validate the consensus rules. In addition they can
+be used by miners to attack their competition, creating perverse incentives, centralization
+pressures and reduced network security.
+
+In computing a block's Merkle root, a 64 bytes transaction can be interpreted as an intermediate
+node in the tree in addition to a leaf. This makes it possible to fake inclusion proofs by
+pretending a 64 bytes block transaction is an inner node, as well as to pretend the inner nodes on
+one level of the tree are the actual block transactions.
+
+Since [bip-0034][BIP34] activation, explicit [bip-0030][BIP30] validation is not necessary until
+block height 1,983,702[^0].  Mandating new coinbase transactions be different from the early
+[bip-0034][BIP34] violations makes it possible to get rid of [bip-0030][BIP30] validation forever.
+Besides its unnecessary cost, another downside of [bip-0030][BIP30] validation is that it cannot be
+performed by Utreexo clients. Finally, leveraging the coinbase transaction's `nLockTime` field
+allows applications to recover the block height corresponding to a coinbase transaction without
+having to parse Bitcoin Script.
+
+# Specification
+
+For all blocks after activation the following new rules apply.
+
+Given a block at height `N`:
+- if `N % 2016` is equal to `0` the `nTime` field of the block must be set to a value higher than or
+  equal to the value of the `nTime` field of block at height `N-1` minus `7200`;
```

Comment:
Maybe reword this to make it clearer that `minus 7200`  is subtracted from the value of `nTime`, and not from `height N-1` (so that no one unfamiliar with the context could read it as  "block at height N - 7201").

Reply:
Is this better?
```suggestion
  equal to the value minus `7200` of the `nTime` field of block at height `N-1`;
```